### PR TITLE
feat(notifications): Notification Settings Page

### PR DESCRIPTION
**Pull request by mgaeta**
Recreate the notification settings page behind a feature flag.
Screen Shot 2021-05-19 at 3 01 48 PM

If this PR is too large I can split it up.

### CODE REVIEW

File: static/app/views/settings/account/notifications/fields2.tsx [hunk 1]
```diff
@@ -18,6 +19,7 @@ export const NOTIFICATION_SETTING_FIELDS: Record<string, NotificationSettingFiel
       ['always', t('On')],
       ['never', t('Off')],
     ],
+    help: 'Enable this to receive alert notifications sent from team alerts.',
```
Comment (scefali) [inline static/app/views/settings/account/notifications/fields2.tsx:22]: These should be translated

File: static/app/views/settings/account/notifications/notificationSettingsByType.tsx [hunk 2]
```diff
@@ -0,0 +1,357 @@
...
+    }
+    return initialData;
+  }
+
+  getFields(): FieldObject[] {
+    const {notificationType} = this.props;
+
+    const fields = [
+      Object.assign(NOTIFICATION_SETTING_FIELDS[notificationType], {
+        help: t('This is the default for all projects.'),
```
Comment (ceorourke) [inline static/app/views/settings/account/notifications/notificationSettingsByType.tsx:268]: I'm seeing this as the text below "Issue alert notifications" always

File: static/app/views/settings/account/notifications/fields2.tsx (hunk 1 above)
Comment (ceorourke) [reply static/app/views/settings/account/notifications/fields2.tsx:22]: I also think this text is a little strange, I can see it's straight from the design but we don't have an outward facing concept of team alerts (aside from what we're building which will live on a different page). Maybe something like "Enable this to receive notifications sent from project alerts"?

File: static/app/views/settings/account/accountNotifications.tsx [hunk 3]
```diff
@@ -34,13 +37,15 @@ const FINE_TUNE_FOOTERS = {
   },
 };
 
-type Props = AsyncView['props'] & {};
+type Props = AsyncView['props'] & {
+  organizations: Organization[];
```
Comment (scefali) [inline static/app/views/settings/account/accountNotifications.tsx:41]: @mgaeta I believe these should actually be typed as `OrganizationSummary`

File: static/app/views/settings/account/notifications/notificationSettingsByType.tsx [hunk 4]
```diff
@@ -0,0 +1,357 @@
...
+    return Object.fromEntries(
+      this.getParents().map(parent => [
+        parent.id,
+        this.getParentValues(parent.id)[provider],
+      ])
+    );
+  };
+
+  getCurrentProviders = (): string[] => {
+    /** Get the list of providers currently active on this page. Note: this can be empty. */
```
Comment (ceorourke) [inline static/app/views/settings/account/notifications/notificationSettingsByType.tsx:129]: When would this be empty?

Comment (mgaeta) [reply static/app/views/settings/account/notifications/notificationSettingsByType.tsx:129]: Yes when the parent-independent notification settings are "never" for all providers this list is empty.

File: static/app/views/settings/account/notifications/notificationSettingsByType.tsx (hunk 2 above)
Comment (mgaeta) [reply static/app/views/settings/account/notifications/notificationSettingsByType.tsx:268]: This looks like `Object.assign` overwriting the constant. :C

File: static/app/views/settings/account/notifications/notificationSettingsByType.tsx [hunk 5]
```diff
@@ -0,0 +1,357 @@
...
+          apiMethod="PUT"
+          apiEndpoint="/users/me/notification-settings/"
+          initialData={this.getInitialData()}
+        >
+          <JsonForm
+            title={this.isGroupedByProject() ? t('All Projects') : t('All Organizations')}
+            fields={this.getFields()}
+          />
+        </Form>
+        {!this.isEverythingDisabled() && (
```
[the remark this answers is not in the document]
Comment (mgaeta) [reply static/app/views/settings/account/notifications/notificationSettingsByType.tsx:330]: Oops I forgot to call the function.

**Review by ceorourke — APPROVED** [review]
Looks good to me, could use another pass with some more seasoned JS eyes though 👀

**Pull request merged**